cmdeploy/src/cmdeploy/opendkim/final.lua

```diff
@@ -21,6 +22,8 @@ for i = 1, nsigs do
 	-- means the message is acceptable.
 	if sigres == 0 then
 		valid = true
+    else
+        sigerrors = sigerrors .. " " .. tostring(sigres)
```

Comment:
I think it's better to create a variable for the error message at the top, set it to `No valid DKIM signature found`. Then on this line replace it with `DKIM signature is invalid, error code 4, search https://github.com/trusteddomainproject/OpenDKIM/blob/master/libopendkim/dkim.h#L108`.

And do not concatenate the error codes, just replace the error. It is unlikely that the sender has multiple DKIM signatures. I just don't trust the ability of OpenDKIM handling long error messages, it is C code that is trivial to segfault from Lua and I don't want to think about what happens if someone sends in a message with 10000 invalid signatures.